src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -392,10 +501,81 @@ def _validate_vmss_create_load_balancer(namespace):
+        raise CLIError('Only one source can be specified: --source-disk | --source-snapshot | '
+                       '--source-blob-uri')
+
+    if namespace.source_disk:
+        namespace.source_disk = _get_ressource_id(namespace.source_disk,
+                                                  namespace.resource_group_name,
+                                                  'disks', 'Microsoft.Compute')
+    elif namespace.source_snapshot:
+        namespace.source_snapshot = _get_ressource_id(namespace.source_snapshot,
+                                                      namespace.resource_group_name,
+                                                      'snapshots', 'Microsoft.Compute')
+
+
+def process_image_create_namespace(namespace):
+    temp = [namespace.os_disk, namespace.os_snapshot,
+            namespace.os_blob_uri, namespace.source_virtual_machine]
+    if sum([1 for t in temp if t]) > 1:
+        raise CLIError('Only one os disk source can be specified: --os-disk | --os-snapshot | '
+                       '--os-blob-uri | --source-virtual-machine')
+
+    if namespace.os_disk:
+        namespace.os_disk = _get_ressource_id(namespace.os_disk, namespace.resource_group_name,
+                                              'disks', 'Microsoft.Compute')
+    if namespace.os_snapshot:
+        namespace.os_snapshot = _get_ressource_id(namespace.os_snapshot,
+                                                  namespace.resource_group_name,
+                                                  'snapshots', 'Microsoft.Compute')
+
+    if namespace.data_disks:
+        ids = []
+        for d in namespace.data_disks:
+            ids.append(_get_ressource_id(d, namespace.resource_group_name,
+                                         'disks', 'Microsoft.Compute'))
+        namespace.data_disks = ids
+
+    if namespace.data_snapshots:
+        ids = []
+        for d in namespace.data_snapshots:
+            ids.append(_get_ressource_id(d, namespace.resource_group_name,
+                                         'snapshots', 'Microsoft.Compute'))
+        namespace.data_snapshots = ids
+
+    if namespace.source_virtual_machine:
+        namespace.source_virtual_machine = _get_ressource_id(namespace.source_virtual_machine,
+                                                             namespace.resource_group_name,
+                                                             'virtualMachines',
+                                                             'Microsoft.Compute')
+        if not namespace.os_type:
+            compute_client = _compute_client_factory()
+            res = parse_resource_id(namespace.source_virtual_machine)
+            vm_info = compute_client.virtual_machines.get(res['resource_group'], res['name'])
+            # pylint: disable=no-member
+            namespace.os_type = vm_info.storage_profile.os_disk.os_type.value
+    elif not namespace.os_type:
+        raise CLIError('Please provide --os-type')
```

Comment:
Recommend a clearer error here. When is --os-type required? "usage error: --something BLAH --os-type OS_TYPE"